### ISSUE TITLE
[BUG] MVGateway and VSCode D3 Rocket

### ISSUE THREAD

**Issue by lecramc** [mvextensions/mvbasic#172]
Hello, everyone!
We previously posted a request as we are allready trying to use VsCode with D3.

using the troubleshooting section of the doc, we tried to connect to mvsp with this Json without success....

POST http://localhost:9005/Login
Content-Type: application/json
{
"AccountName":"dm",    
"ServerType": "D3",  
"RemoteHost": "http://192.168.2.4:9000/",
"ServerName":"",
"UserId" : "dm",
 "Password" : ""
}

we always have a 500 error
{
  "Message": "An error has occurred.",
  "ExceptionMessage": "La référence d'objet n'est pas définie à une instance d'un objet.",
  "ExceptionType": "System.NullReferenceException",
  "StackTrace": "   à UnifiedObjects.ServerConnection..ctor(NetbuilderServer server, String accountName)\r\n   à MvonGateway.MvonGateway.findConnection(String accountName)\r\n   à [MvonGateway.LoginController.Post](http://mvongateway.logincontroller.post/)(LoginDetails value)\r\n  <snip>

**To Reproduce**
We just use tutorial in read me project.

![image](https://user-images.githubusercontent.com/35737100/202995800-51075130-4e20-4566-9a3b-b7a6915be113.png)

**Screenshots**

**Versions of applicable software**
Detail on the versions of any and all pieces of software that are part of your environment, such as:
  Host : Windows 11 or windows 10
  Remote : Red Hat Enterprise Linux 8.3 (Ootpa)"
  MV Basic extension
  Vscode 1.73

I think is a problem with mv gateway, we can't even get a test connection to the remote server

**Comment by itsxallwater** [comment]
Hey @lecramc thanks for reporting. Would you have the ability to try one of the other connector options? Otherwise perhaps @dthiot or @GrantHart could chime in with tricks for D3. This may also come down to general compatibility of the extension with the MV Gateway connector.

**Comment by lecramc** [comment]
Thank's @itsxallwater, yes maybe the second option (accuterm), but if we can avoid to pay...
Ok maybe try an older version of mvbasic extension? 

Rocket's extension doesn't allowed D3 server if I'm not mistaken...

**Comment by dthiot** [comment]
Either Grant H or myself will look into this sometime this week.

Dick

From: Clément MARCEL ***@***.***>
Date: Monday, November 21, 2022 at 11:28 AM
To: mvextensions/mvbasic ***@***.***>
Cc: Dick Thiot ***@***.***>, Mention ***@***.***>
Subject: Re: [mvextensions/mvbasic] [BUG] MVGateway and VSCode D3 Rocket (Issue #172)

Thank's @itsxallwater<https://github.com/itsxallwater>, yes maybe the second option (accuterm), but if we can avoid to pay...
Ok maybe try an older version of mvbasic extension?

Rocket's extension doesn't allowed D3 server if I'm not mistaken...

**Comment by lecramc** [comment]
Ok thank's @dthiot ! if I can help, I am at your disposal

**Comment by dthiot** [comment]
Please send me your contact information to ***@***.******@***.***>.  Are you in France?

Dick

From: Clément MARCEL ***@***.***>
Date: Monday, November 21, 2022 at 12:44 PM
To: mvextensions/mvbasic ***@***.***>
Cc: Dick Thiot ***@***.***>, Mention ***@***.***>
Subject: Re: [mvextensions/mvbasic] [BUG] MVGateway and VSCode D3 Rocket (Issue #172)

Ok thank's @dthiot<https://github.com/dthiot> ! if I can help, I am at your disposal

**Comment by lecramc** [comment]
Yes I'm form France, I setd an email to your mail address on your github's profile.

**Comment by lecramc** [comment]
Hi everyone, 
Today we have test with accuterm, that's work fine.

But we have a lot of "subfolder", can we list juste one of them or more ? 

Best regards,

**Comment by dthiot** [comment]
Clément,

Yes there is a way to limit the files but I don’t remember how that was implemented for Accuterm.  Hopefully, Peter or Mike can reply.  Later today, if I don’t see a reply I will dig into my system and find what it is.

Dick

From: Clément MARCEL ***@***.***>
Date: Tuesday, November 29, 2022 at 8:53 AM
To: mvextensions/mvbasic ***@***.***>
Cc: Dick Thiot ***@***.***>, Mention ***@***.***>
Subject: Re: [mvextensions/mvbasic] [BUG] MVGateway and VSCode D3 Rocket (Issue #172)

Hi everyone,
Today we have test with accuterm, that's work fine.

But we have a lot of "subfolder", can we list juste one of them or more ?

Best regards,

**Comment by lecramc** [comment]
Thank's Dick, 
I have tried with MvBasic.EditFiles option but I'm not sure if this is the right way or I haven't the right synthax.
Maybe we will take the license for accuterm after the trial version but if you have any feedback on MvGateway and my mistake I am interested.

Thank's a lot !

**Comment by pschellenbach** [comment]
Clément,

There is a way to specify a list of files to return when VSCode requests the "root" directory of the file system. When using the AccuTerm connector, the "root" directory maps to the account MD or VOC. The default is to return a list of all the files found in MD/VOC. In some environments this list can be quite long (and time consuming to scan). 

There is a small note about using MVCONNECTION.OPTIONS to restrict the "root" file list in the AccuTerm release notes, just search for "MVCONNECTION.OPTIONS". Also, more information about this feature is available here [https://github.com/pickmultivalue/accuterm-host-programs/issues/2](https://github.com/pickmultivalue/accuterm-host-programs/issues/2). 

Hope this helps,

Pete

**Comment by lecramc** [comment]
Hi Pete, 

Thank you! It works fine, we can filter on the file name.

Clément
